com.unity.cinemachine/Tests/Runtime/CmColliderTests.cs

```diff
@@ -130,40 +115,49 @@ public IEnumerator CheckDamping()
+            yield return WaitForOnePhysicsFrame(); // ensure that moving the collider (obstacle) takes effect
             
+            // 2. Check that camera moved (no damping, so it snapped).
+            yield return UpdateCinemachine();
+            Assert.That(originalCamPosition, Is.Not.EqualTo(m_Vcam.State.GetFinalPosition()).Using(m_Vector3EqualityComparer));
             var obstructedPosition = m_Vcam.State.GetFinalPosition();
-            Assert.That(originalCamPosition, !Is.EqualTo(obstructedPosition).Using(Vector3EqualityComparer.Instance));
-
-            // wait another frame to avoid snap - we need to have a previous damp time to avoid snap
-            yield return null;
-            m_Brain.ManualUpdate();
-            Assert.That(obstructedPosition, Is.EqualTo(m_Vcam.State.GetFinalPosition()).Using(Vector3EqualityComparer.Instance));
             
-            UnityEngine.Object.Destroy(obstacle);
+            // 3. Check that camera still has not moved 
+            yield return UpdateCinemachine(); // wait another frame to avoid snap - we need to have a previous damp time to avoid snap
+            Assert.That(obstructedPosition, Is.EqualTo(m_Vcam.State.GetFinalPosition()).Using(m_Vector3EqualityComparer));
             
-            yield return WaitForOnePhysicsFrame();
-            m_Brain.ManualUpdate();
+            // 4. Remove obstacle
+            yield return PhysicsDestroy(obstacle);
             
-            // camera has moved and it is not yet back at its original position because damping > 0.
-            var finalPosition = m_Vcam.State.GetFinalPosition();
-            Assert.That(obstructedPosition, Is.Not.EqualTo(finalPosition).Using(Vector3EqualityComparer.Instance));
-            Assert.That(originalCamPosition, Is.Not.EqualTo(finalPosition).Using(Vector3EqualityComparer.Instance));
-            Assert.That(new Vector3(0, 0, -4.71081734f), Is.EqualTo(finalPosition).Using(Vector3EqualityComparer.Instance));
+            // 5. Wait for correction damping as the camera is returning to the starting position and test damping.
+            yield return WaitForDamping();
+
+            // 6. Check that camera is back to original position
+            yield return UpdateCinemachine();
+            Assert.That(originalCamPosition, Is.EqualTo(m_Vcam.State.GetFinalPosition()).Using(m_Vector3EqualityComparer));
+        }
+        
+        IEnumerator WaitForDamping()
+        {
+            var previousDelta = -1f;
+            do
+            {
+                var startPosition = m_Vcam.State.GetFinalPosition();
+                yield return UpdateCinemachine();
+                var delta = (m_Vcam.State.GetFinalPosition() - startPosition).sqrMagnitude;
+                Assert.That(delta, Is.Not.Negative);
+                if (previousDelta >= 0)
+                    Assert.That(delta, Is.LessThanOrEqualTo(previousDelta / 2f));
```

Comment:
"It should take the approximately the damping time to get to the destination". This is not the case currently.

DampingWhenOccluded takes 1.8 sec, when DampingWhenOccluded is 1, to go from -5 to -4.5.
Damping takes 0.6 sec, when Damping is 1, to go from -4.5 to -5.

Reply:
That sounds like a problem waiting to get fixed.  Why is it not taking the correct time?

Reply:
Checking

Reply:
I have a vague memory of there being damping tests once upon a time in UnityVectorExtensionTests.  Maybe it was just a dream.

Reply:
I checked the code. The reason why damping doesn't behave the same is because we do some extra damping on the dampTime in CmDeoccluder.
```
if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid && AvoidObstacles.DampingWhenOccluded + AvoidObstacles.Damping > Epsilon)
{
    // To ease the transition between damped and undamped regions, we damp the damp time
    var dispSqrMag = displacement.sqrMagnitude;
    dampTime = dispSqrMag > extra.PreviousDisplacement.sqrMagnitude ? AvoidObstacles.DampingWhenOccluded : AvoidObstacles.Damping;
    if (dispSqrMag < Epsilon)
        dampTime = extra.PreviousDampTime - Damper.Damp(extra.PreviousDampTime, dampTime, deltaTime);

    var prevDisplacement = state.ReferenceLookAt + dampingBypass * extra.PreviousCameraOffset - initialCamPos;
    displacement = prevDisplacement + Damper.Damp(displacement - prevDisplacement, dampTime, deltaTime);
}
```

The difference in damp times that I mentioned above (
"DampingWhenOccluded takes 1.8 sec, when DampingWhenOccluded is 1, to go from -5 to -4.5.
Damping takes 0.6 sec, when Damping is 1, to go from -4.5 to -5.")
is due the `if (dispSqrMag < Epsilon)` check. 
When we are occluded (DampingWhenOccluded), PreserveLineOfSight sets displacement to a non-zero vector, whereas when we are not occluded (Damping), displacement is always a zero vector.
`var displacement = PreserveLineOfSight(ref state, ref extra);`

Checking `(dispSqrMag < Epsilon)` may be a wrong and cause of a bug, but we may have wanted it this way. I don't remember.
If its a bug, then we could maybe use extra.PreviousCameraOffset or prevDisplacement instead to calculate displacement magnitude:
`var prevDisplacement = state.ReferenceLookAt + dampingBypass * extra.PreviousCameraOffset - initialCamPos;`

Reply:
I added Vector3 DampingTests. 

Time is consistent with what we saw in the Deoccluder tests:
"DampingWhenOccluded takes 1.8 sec, when DampingWhenOccluded is 1, to go from -5 to -4.5."